entity/src/content_audit.rs

```diff
@@ -335,6 +376,56 @@ pub async fn get_audits<T: OverlayContentKey>(
         .await?)
 }
 
+pub async fn get_failed_keys(
+    subprotocol: SubProtocol,
+    strategy_used: String,
+    page: u32,
+    conn: &DatabaseConnection,
+) -> Result<Vec<String>> {
+    let page_size: u32 = 1000;
+
+    let subprotocol_strategy: SelectionStrategy = match subprotocol {
+        SubProtocol::History => SelectionStrategy::History(strategy_used.try_into()?),
+        SubProtocol::State => SelectionStrategy::State(strategy_used.try_into()?),
+        SubProtocol::Beacon => SelectionStrategy::Beacon(strategy_used.try_into()?),
+    };
+
+    let keys_result = FailedKeysResult::find_by_statement(Statement::from_sql_and_values(
+        DbBackend::Postgres,
+        "
+        SELECT
```

Comment:
This query is equivalent (I verified via `explain analyze` that it generates the same plan and runs the same) and I think a little more readable without the sub-query:
```suggestion
        SELECT
              content.content_key
        FROM content_audit
        INNER JOIN content ON content.id = content_audit.content_key
        WHERE 
            content_audit.result = 0 AND
            content_audit.strategy_used = $1 AND
            content_audit.created_at >= CURRENT_TIMESTAMP - INTERVAL '24 hours'
        GROUP BY content.content_key
        ORDER BY MAX(content_audit.created_at) DESC
        LIMIT $2
        OFFSET $3
```